tests/Cms/Files/FileRulesTest.php

```diff
@@ -113,6 +122,69 @@ public function testCreateExistingFile()
         FileRules::create($file, $upload);
     }
 
+    public function testCreateSameFile()
+    {
+        $testImage =  __DIR__ . '/fixtures/files/test.jpg';
+
+        $app = new App([
+            'roots' => [
+                'index' => $this->fixtures = __DIR__ . '/fixtures/FileRulesTest/createSameFile',
+            ],
+            'site' => [
+                'children' => [
+                    [
+                        'slug' => 'test',
+                        'files' => [
+                            ['filename' => 'test.jpg', 'content' => ['template' => 'test']],
+                        ]
+                    ]
+                ]
+            ]
+        ]);
+
+        $page = $app->page('test');
+
+        // create real file with content and move into page root
+        F::copy($testImage, $page->root() . '/test.jpg');
+        F::write($page->root() . '/test.jpg.txt', 'Template: test');
+
+        // create new file
+        $newFile = new File([
+            'filename' => 'test.jpg',
+            'parent' => $page,
+            'content' => [
+                'template' => 'test'
+            ]
+        ]);
+
+        // same file
+        $upload = new BaseFile($testImage);
+        $create = FileRules::create($newFile, $upload);
+        $this->assertTrue($create);
+
+        // same file and different template
+        $this->expectException('Kirby\Exception\DuplicateException');
+        $this->expectExceptionMessage('A file with the name "test.jpg" already exists');
+
+        $newFile = new File([
+            'filename' => 'test.jpg',
+            'parent' => $page,
+            'content' => [
+                'template' => 'cover'
+            ]
+        ]);
+
+        $upload = new BaseFile($testImage);
+        FileRules::create($newFile, $upload);
+
+        // same filename, same template and different file (sha1)
```

Comment:
I think this needs to be in a separate method as the method ends when the first exception is thrown.

Reply:
So do we need:

- 3 unit tests? For `same file`, `same file and different template` and `same filename, same template and different file (sha1)`? 

- 2 unit tests? For `true` and `exceptions`.

Reply:
Maybe best to put them in three methods, yes